Analysis/Tasks/PWGHF/HFCandidateCreator3Prong.cxx

```diff
@@ -170,18 +182,38 @@ struct HFCandidateCreator3ProngMC {
       // Λc± → p± K∓ π±
       if (result == N3ProngDecays) {
         //Printf("Checking Λc± → p± K∓ π±");
-        if (RecoDecay::getMatchedMCRec(particlesMC, std::move(arrayDaughters), 4122, array{+kProton, -kKPlus, +kPiPlus}, true, &sign) > -1) {
+        auto indexRecLc = RecoDecay::getMatchedMCRec(particlesMC, std::move(arrayDaughters), 4122, array{+kProton, -kKPlus, +kPiPlus}, true, &sign, -1);
+        if (indexRecLc > -1) {
           result = sign * LcToPKPi;
+
+          //Printf("Flagging the different Λc± → p± K∓ π± decay channels");
+          RecoDecay::getDaughters(particlesMC, indexRecLc, &arrDaughIndex, array{+kProton, -kKPlus, +kPiPlus}, 1);
+          size = arrDaughIndex.size();
+          for (auto iProng = 0; iProng < size; ++iProng) {
+            auto daughI = particlesMC.iteratorAt(arrDaughIndex[iProng]);
+            arrPDGDaugh[iProng] = daughI.pdgCode();
+          }
+          if (size == 2) {
+            if ((std::abs(arrPDGDaugh[0]) == arrPDGResonant1[0] && std::abs(arrPDGDaugh[1]) == arrPDGResonant1[1]))
```

Comment:
If you assign directly `arrPDGDaugh[iProng] = std:abs(daughI.pdgCode());`, you don't have to repeat the `std:abs` later anymore.